File: backend/src/main/scala/models/ChatMessage.scala
package models

import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

case class ChatMessage(id: Int, chatMemberId: Int, message: String) {
}
object ChatMessage {
  implicit val encoder = deriveEncoder[ChatMessage]
  implicit val decoder = deriveDecoder[ChatMessage]
}

File: frontend/src/app/models/ChatMessage.ts
export class ChatMessage {
  id: number;
  chatMemberId: number;
  message: string;

  constructor(id: number, chatMemberId: number, message: string) {
    this.id = id;
    this.chatMemberId = chatMemberId;
    this.message = message;
  }
}
